dlk/tests/test_code_generation.py

```diff
@@ -90,83 +90,145 @@ def dict_codegen_segmentation(cpu_name) -> dict:
             }
 
 
-def get_configurations_by_test_cases(test_cases, configuration):
-
-    return [updated_dict(configuration,test_case) for test_case in test_cases]
-
-
-def get_configurations_by_architecture(test_cases, cpu_name):
-    configurations = []
-    configurations.extend(get_configurations_by_test_cases(test_cases, dict_codegen_classification(cpu_name)))
-    configurations.extend(get_configurations_by_test_cases(test_cases, dict_codegen_classification_resnet(cpu_name)))
-    configurations.extend(get_configurations_by_test_cases(test_cases, dict_codegen_object_detection(cpu_name)))
-    configurations.extend(get_configurations_by_test_cases(test_cases, dict_codegen_segmentation(cpu_name)))
-
-    return configurations
-
-
 def get_configurations_x86_64():
     cpu_name = "x86_64"
-    test_cases = [
-        {'use_avx': True, 'hard_quantize': True, 'threshold_skipping': True},
-        {'use_avx': True, 'hard_quantize': True, 'threshold_skipping': False},
-        {'use_avx': True, 'hard_quantize': False, 'threshold_skipping': False},
-        {'use_avx': False, 'hard_quantize': True, 'threshold_skipping': True},
-        {'use_avx': False, 'hard_quantize': True, 'threshold_skipping': False},
-        {'use_avx': False, 'hard_quantize': False, 'threshold_skipping': False},
-    ]
-    configurations = get_configurations_by_architecture(test_cases, cpu_name)
-
-    additional_test_configuration = updated_dict(dict_codegen_classification(cpu_name),
-                                                 {'use_run_test_script': True})
-    additional_test_cases = [
-        {'use_avx': True, 'input_name': 'raw_image.png', 'test_level': TEST_LEVEL_FUTURE_TARGET},
-        {'use_avx': True, 'input_name': 'preprocessed_image.npy', 'from_npy': True},
-        {'use_avx': False, 'input_name': 'raw_image.png', 'test_level': TEST_LEVEL_FUTURE_TARGET},
-        {'use_avx': False, 'input_name': 'preprocessed_image.npy', 'from_npy': True},
+    cls_base = dict_codegen_classification(cpu_name)
+    cls_resnet_base = dict_codegen_classification_resnet(cpu_name)
+    det_base = dict_codegen_object_detection(cpu_name)
+    seg_base = dict_codegen_segmentation(cpu_name)
+    configurations = [
```

Comment:
This look overly complex.
What about
```
sub_patterns = [{'use_avx': True, ...}, ...]
configurations = []
configurations.extend(get_configurations_by_test_cases(cls_base, sub_patterns)
...

return list(enumerate(configurations))
```

Well, this also look complex :(
do you have better idea?